include/deal.II/multigrid/mg_transfer.h

```diff
@@ -102,6 +102,21 @@ namespace internal
     }
   };
 #endif
+
+  /**
+   * A data structure that holds specialized structures needed for
+   * MGTransfer<parallel::distributed::Vector<Number> >.
+   */
+  template <typename Number>
+  struct ParallelVectorData
+  {
+    mutable parallel::distributed::Vector<Number> ghosted_global_vector;
+    mutable MGLevelObject<parallel::distributed::Vector<Number> > ghosted_level_vector;
+    std::vector<std::vector<std::pair<unsigned int,unsigned int> > > copy_indices;
+    std::vector<std::vector<std::pair<unsigned int,unsigned int> > > copy_indices_level_mine;
+    std::vector<std::vector<std::pair<unsigned int,unsigned int> > > copy_indices_global_mine;
+    bool perform_plain_copy;
```

Comment:
I really like the plain_copy optimization. This would probably also be neat for the PETSc/Trilinos case, right? Would it be a lot of effort to move this bool outside of ```data```?